components/collator/src/elements.rs

```diff
@@ -157,9 +156,8 @@ const SINGLE_REPLACEMENT_CHARACTER_U16: &ZeroSlice<u16> =
 
 pub(crate) const EMPTY_CHAR: &ZeroSlice<char> = ZeroSlice::new_empty();
 
-const SINGLE_REPLACEMENT_CHARACTER_CHAR: &ZeroSlice<char> = ZeroSlice::from_ule_slice(&[unsafe {
-    core::mem::transmute::<[u8; 3], CharULE>([0xFDu8, 0xFFu8, 0u8])
-}]);
+const SINGLE_REPLACEMENT_CHARACTER_CHAR: &ZeroSlice<char> =
+    ZeroSlice::<char>::from_ule_slice(&<char as AsULE>::ULE::from_array([REPLACEMENT_CHARACTER]));
```

Comment:
This is still quite the expression. You could take a stab at further simplifying this with https://github.com/unicode-org/icu4x/issues/1935 if you want.

Reply:
Since the `zeroslice!` macro doesn't exist yet, this is what #1935 allows me to do if I'm not mistaken. I picked the `<char as AsULE>::ULE` way to refer to `CharULE` to be consistent with the lines above (`<u16 as AsULE>`), but I could switch to simply `CharULE` if you want? I don't have a preference.

Reply:
I meant you can implement the `zeroslice!` macro 😉 

`<char as AsULE>::ULE`, while longer, is nicer because the ULE types are implementation details.

Reply:
Ah! Got it 😄 I'll open a new PR